diff --git a/src/installer.js b/src/installer.js
index a2c427840576e0b41853b0be913b04e9766763d1..f7fc2dfb3d17b867a0e57787fe23a7c4cb381bca 100644
--- a/src/installer.js
+++ b/src/installer.js
@@ -103,6 +103,7 @@ class FlatpakInstaller extends common.ElectronInstaller {
       ...(await this.determineBaseRuntimeAndSDK()),
       id: getAppID(pkg.name, common.getHomePage(pkg)),
       extraFlatpakBuilderArgs: [],
+      extraFlatpakBuildExportArgs: [],
       finishArgs: [
         // X Rendering
         '--socket=x11', '--share=ipc',
@@ -175,6 +176,7 @@ class FlatpakInstaller extends common.ElectronInstaller {
       bundlePath: dest,
       extraExports: extraExports,
       extraFlatpakBuilderArgs: this.options.extraFlatpakBuilderArgs,
+      extraFlatpakBuildExportArgs: this.options.extraFlatpakBuildExportArgs,
       files: files.concat(this.options.files),
       symlinks: symlinks.concat(this.options.symlinks)
     })
