diff --git a/dist/server/config.js b/dist/server/config.js
index 3222d6521a5473994ad1d42914106d9c611e18b8..b9ea8fbd541c9ce1a8c33c25ab0f493981f9b3e3 100644
--- a/dist/server/config.js
+++ b/dist/server/config.js
@@ -180,8 +180,8 @@ const isAboveNodejs16 = (0, _semver).gte(process.version, NODE_16_VERSION);
 const isAboveNodejs18 = (0, _semver).gte(process.version, NODE_18_VERSION);
 const experimentalWarning = (0, _utils).execOnce((configFileName, features)=>{
     const s = features.length > 1 ? "s" : "";
-    Log.warn(_chalk.default.bold(`You have enabled experimental feature${s} (${features.join(", ")}) in ${configFileName}.`));
-    Log.warn(`Experimental features are not covered by semver, and may cause unexpected or broken application behavior. ` + `Use at your own risk.`);
+    // Log.warn(_chalk.default.bold(`You have enabled experimental feature${s} (${features.join(", ")}) in ${configFileName}.`));
+    // Log.warn(`Experimental features are not covered by semver, and may cause unexpected or broken application behavior. ` + `Use at your own risk.`);
     if (features.includes("appDir")) {
         Log.info(`Thank you for testing \`appDir\` please leave your feedback at https://nextjs.link/app-feedback`);
     }
